src/main/java/frc/robot/subsystems/snapback/hood/HoodIOTalonFX.java

```diff
@@ -1,3 +1,93 @@
 package frc.robot.subsystems.snapback.hood;
 
-public class HoodIOTalonFX {}
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+import com.ctre.phoenix6.controls.VoltageOut;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.units.measure.Angle;
+import edu.wpi.first.units.measure.AngularVelocity;
+import edu.wpi.first.units.measure.Current;
+import edu.wpi.first.units.measure.Temperature;
+import frc.robot.util.Alert;
+import frc.robot.util.Alert.AlertType;
+
+public class HoodIOTalonFX implements HoodIO {
+  private final TalonFX hoodMotor;
+  public StatusSignal<Angle> positionRotations;
+  public StatusSignal<AngularVelocity> velocityRotationsPerSecond;
+  public StatusSignal<Current> currentAmps;
+  public StatusSignal<Temperature> tempratureCelsius;
+  public StatusSignal<Double> positionSetpointRotationsPerSecond;
+  public StatusSignal<Double> positionErrorRotationsPerSecond;
+
+  private final Alert disconnectedAlert =
+      new Alert("Hood Talon is disconnected, check CAN bus.", AlertType.ERROR);
+
+  private VoltageOut voltageControlRequest;
+  private MotionMagicVoltage positionControlRequest;
+
+  public HoodIOTalonFX() {
+    hoodMotor = new TalonFX(HoodConstants.HOOD_MOTOR_CAN_ID);
+
+    TalonFXConfiguration config = new TalonFXConfiguration();
+    config.CurrentLimits.SupplyCurrentLimit = 60.0;
+    config.CurrentLimits.SupplyCurrentLimitEnable = true;
+    config.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+    config.Feedback.SensorToMechanismRatio = HoodConstants.GEAR_REDUCTION;
+    config.Slot0.kP = HoodConstants.GAINS.kp();
+    config.Slot0.kI = HoodConstants.GAINS.ki();
+    config.Slot0.kD = HoodConstants.GAINS.kd();
+    config.Slot0.kS = HoodConstants.GAINS.ks();
+    config.Slot0.kV = HoodConstants.GAINS.kv();
+    config.Slot0.kA = HoodConstants.GAINS.ka();
+    config.MotionMagic.MotionMagicAcceleration = HoodConstants.GAINS.ka();
+    config.MotionMagic.MotionMagicCruiseVelocity = HoodConstants.GAINS.kv();
```

Comment:
kv is the velocity gain for the feedforward controller, not the motion profile max velocity, we need a constant for the max velocity of the motion profile.